pymacaroons/serializers/binary_serializer.py

```diff
@@ -1,19 +1,44 @@
 from __future__ import unicode_literals
 
 import binascii
+from collections import namedtuple
+import six
 import struct
-from base64 import urlsafe_b64encode, urlsafe_b64decode
+import sys
+from base64 import urlsafe_b64encode
 
-from pymacaroons.utils import convert_to_bytes
+from pymacaroons.utils import (
+    convert_to_bytes,
+    convert_to_string,
+    raw_b64decode,
+)
 from pymacaroons.serializers.base_serializer import BaseSerializer
 from pymacaroons.exceptions import MacaroonSerializationException
 
+_LOCATION = 1
+_IDENTIFIER = 2
+_VID = 4
+_SIGNATURE = 6
+_EOS = 0
+
+PacketV2 = namedtuple('PacketV2', ['field_type', 'data'])
+
 
 class BinarySerializer(BaseSerializer):
 
     PACKET_PREFIX_LENGTH = 4
 
     def serialize(self, macaroon):
+        return urlsafe_b64encode(
+            self.serialize_raw(macaroon)).decode('ascii').rstrip('=')
+
+    def serialize_raw(self, macaroon):
```

Comment:
In my experience, deserialization is the time when you generally *don't* know what version to expect, unless you send the version out of band, which seems unnecessary given that we can determine the version by inspecting the encoded macaroon.

For example, consider a web server that takes a macaroon as authentication in its request header. The web server might have be serving macaroons to clients with different versions, and therefore be issuing different macaroon versions, but each client will present their macaroon in the same way.